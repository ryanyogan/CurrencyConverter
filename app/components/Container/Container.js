import React from 'react';
import { any } from 'prop-types';
import { View } from 'react-native';

import styles from './styles';

const Container = ({ children }) => (
  <View style={styles.container}>{children}</View>
);

Container.propTypes = {
  children: any // eslint-disable-line
};

export default Container;
