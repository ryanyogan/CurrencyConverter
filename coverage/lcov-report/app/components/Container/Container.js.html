<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/components/Container/Container.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">app/components/Container</a> Container.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">85.71% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>6/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">85.71% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/7</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import { any, string } from 'prop-types';
import { View, TouchableWithoutFeedback, Keyboard } from 'react-native';
&nbsp;
import styles from './styles';
&nbsp;
const Container = ({ children, backgroundColor }) =&gt; {
  const containerStyles = [styles.container];
  if (backgroundColor) {
    containerStyles.push({ backgroundColor });
  }
&nbsp;
  return (
    &lt;TouchableWithoutFeedback onPress={<span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >Keyboard.dismiss()}</span>&gt;
      &lt;View style={containerStyles}&gt;{children}&lt;/View&gt;
    &lt;/TouchableWithoutFeedback&gt;
  );
};
&nbsp;
Container.propTypes = {
  children: any, // eslint-disable-line
  backgroundColor: string,
};
&nbsp;
export default Container;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Feb 05 2018 09:50:31 GMT-0800 (PST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
