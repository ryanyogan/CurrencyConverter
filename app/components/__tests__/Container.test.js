import React from 'react';
import renderer from 'react-test-renderer';
import { View, Keyboard } from 'react-native';

import { Container, styles } from '../Container';

it('renders the Container component', () => {
  const rendered = renderer.create(<Container />).toJSON();
  expect(rendered).toBeTruthy();
});

it('styles is an object', () => {
  expect(typeof styles).toBe('object');
});

it('renders the child prop', () => {
  const rendered = renderer
    .create(
      <Container>
        <View />
      </Container>,
    )
    .toJSON();
  expect(rendered).toMatchSnapshot();
});

it('uses the specified backgroundColor, if prop is provided', () => {
  const rendered = renderer
    .create(<Container backgroundColor="red" />)
    .toJSON();
  expect(rendered).toMatchSnapshot();
});

it('dismisses the keyboard onPress', () => {
  const rendered = renderer
    .create(<Container onPress={() => Keyboard.dismiss()} />)
    .toJSON();
  expect(rendered).toMatchSnapshot();
});
