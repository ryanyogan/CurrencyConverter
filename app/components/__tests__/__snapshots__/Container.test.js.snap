// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dismisses the keyboard onPress 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      undefined,
    ]
  }
  testID={undefined}
/>
`;

exports[`renders the child prop 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      undefined,
    ]
  }
  testID={undefined}
>
  <View />
</View>
`;

exports[`uses the specified backgroundColor, if prop is provided 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "red",
      },
    ]
  }
  testID={undefined}
/>
`;
